XIPs/xip-36-frames-support.md

```diff
@@ -0,0 +1,253 @@
+
+These payloads are meaningfully different from the Farcaster payloads. We can provide some helper libraries to allow a Frames developer to take the JSON payload above and get back the information required to move forward to the next Frame (`isValid`, `verifiedSenderWalletAddress`, `buttonIndex`). Because XMTP is a private network instead of a public one, the goal should be to include the minimum amount of metadata to make this work.
+
+On the sender side (the person clicking the button), XMTP SDKs already have generic interfaces for signing messages. All we need to do is define the payload format, and any app developer should be able to generate a payload matching this format and transmit it directly to the Frame URL.
+
+The `FrameAction` includes a signature of the `FrameActionBody` encoded as bytes. To verify and recover the wallet address from this payload, a developer must:
+
+1. Recover the public key from the signature
+2. Verify that the public key matches the identity key in the `SignedPublicKeyBundle`
+3. Recover the wallet address from the signature in the `SignedPublicKeyBundle`
+
+For a prototype implementation of a verification library that does this, see the [frames-validator](https://github.com/xmtp/xmtp-node-js-tools/tree/main/packages/frames-validator) in the **xmtp-node-js-tools** GitHub repo.
+
+## Backward compatibility
+
+- All Frames will be sent using ordinary XMTP text messages.
+- Client apps can choose to expand URLs found in these messages into Frames at their discretion.
+- No new Content Types are required.
+- No new SDK changes are necessary for XMTP SDKs that allow app developers to sign messages with the XMTP identity key.
+- The new helper libraries that facilitate Frame rendering, interaction, and verification are strictly opt-in.
+
+## Reference implementations
+
+We've developed reference implementations for all three components required to enable Frames on XMTP:
+
+1. [Rendering/interaction helper library](https://github.com/xmtp/xmtp-web/tree/main/packages/frames-client)  
+Needs some updates to accommodate changes to the POST message schema made after development began
+
+2. [POST payload verification helper library](https://github.com/xmtp/xmtp-node-js-tools/tree/main/packages/frames-validator)  
+Needs some updates to accommodate changes to the POST message schema made after development began
+
+3. [Open Graph Proxy service](https://github.com/neekolas/og-proxy)  
+Still missing support for proxying requests for images. This support will be a hard requirement for launch.
+
+Here are some specifications you might also want to explore as a part of working with these reference implementations:
```

Comment:
See below - this is where I added the link to the Open Frames spec as suggested by Nick. 

I also added a link to the Farcaster Frames spec, just to be comprehensive.